### PR TITLE
[SYCL] MacOS support for os_util

### PR DESCRIPTION
**Pull request by jeffhammond**
Slight modifications to the original implementation by @alexbatashev.

**These are sufficient for a clean build on MacOS** and I encourage them to be merged soon.  Rebasing Alex's patches is hard given the rate of change in the sycl branch.

Signed-off-by: Hammond, Jeff R <jeff.r.hammond@intel.com>
Signed-off-by: Alexander Batashev <alexander.batashev@intel.com> (please re-affirm the sign-off that was in your original patch set)

### CODE REVIEW

File: sycl/source/detail/os_util.cpp [hunk 1]
```diff
@@ -93,19 +113,27 @@ size_t OSUtil::getOSMemSize() {
...
+  return static_cast<size_t>(Size);
+#endif // SYCL_RT_OS
 }
 
 void *OSUtil::alignedAlloc(size_t Alignment, size_t NumBytes) {
 #if defined(SYCL_RT_OS_LINUX)
   return aligned_alloc(Alignment, NumBytes);
+#elif defined(SYCL_RT_OS_POSIX_SUPPORT)
+  void *Addr;
+  int rc = posix_memalign(&Addr, Alignment, NumBytes);
```
Comment (alexbatashev) [inline sycl/source/detail/os_util.cpp]: Small nit
```suggestion
  int ReturnCode = posix_memalign(&Addr, Alignment, NumBytes);
```

File: sycl/source/detail/os_util.cpp [hunk 2]
```diff
@@ -93,19 +113,27 @@ size_t OSUtil::getOSMemSize() {
...
+#endif // SYCL_RT_OS
 }
 
 void *OSUtil::alignedAlloc(size_t Alignment, size_t NumBytes) {
 #if defined(SYCL_RT_OS_LINUX)
   return aligned_alloc(Alignment, NumBytes);
+#elif defined(SYCL_RT_OS_POSIX_SUPPORT)
+  void *Addr;
+  int ReturnCode = posix_memalign(&Addr, Alignment, NumBytes);
+  return (ReturnCode==0) ? Addr : nullptr;
```
Comment (AGindinson) [inline sycl/source/detail/os_util.cpp]: Is this in accordance with clang-format?

Comment (jeffhammond) [reply sycl/source/detail/os_util.cpp]: Fixed by 4c52bee2521aa42c9ed993e98e3d5c311f8678c5

File: sycl/source/detail/os_util.cpp [hunk 3]
```diff
@@ -93,19 +112,27 @@ size_t OSUtil::getOSMemSize() {
...
+  sysctlbyname("hw.memsize", &Size, nullptr, nullptr, 0);
+  return static_cast<size_t>(Size);
+#endif // SYCL_RT_OS
 }
 
 void *OSUtil::alignedAlloc(size_t Alignment, size_t NumBytes) {
 #if defined(SYCL_RT_OS_LINUX)
   return aligned_alloc(Alignment, NumBytes);
+#elif defined(SYCL_RT_OS_POSIX_SUPPORT)
+  void *Addr;
```
Comment (romanovvlad) [inline sycl/source/detail/os_util.cpp]:
```suggestion
  void *Addr = nullptr;
```

File: sycl/source/detail/os_util.cpp [hunk 4]
```diff
@@ -93,19 +112,27 @@ size_t OSUtil::getOSMemSize() {
...
+  int ReturnCode = posix_memalign(&Addr, Alignment, NumBytes);
+  return (ReturnCode == 0) ? Addr : nullptr;
 #elif defined(SYCL_RT_OS_WINDOWS)
   return _aligned_malloc(NumBytes, Alignment);
 #endif
 }
 
 void OSUtil::alignedFree(void *Ptr) {
-#if defined(SYCL_RT_OS_LINUX)
+#if defined(SYCL_RT_OS_LINUX) || defined(SYCL_RT_OS_POSIX_SUPPORT)
```
Comment (romanovvlad) [inline sycl/source/detail/os_util.cpp]: Isn't checking for SYCL_RT_OS_POSIX_SUPPORT is enough?

**Comment by romanovvlad** [comment]
There is unresolved comment.

**Comment by jbrodman** [comment]
I think a commit is missing being signed (-s)

**Comment by jeffhammond** [comment]
@jbrodman Attempts to sign the more recent commits is proving impossible.  Suggestions?
```
jrhammon-mac02:llvm jrhammon$ git commit --amend efd598fdd00e7aad7c19876e09edae32f6b10a44
error: pathspec 'efd598fdd00e7aad7c19876e09edae32f6b10a44' did not match any file(s) known to git
```

**Comment by jeffhammond** [comment]
I just squashed those commits into the main one.

File: sycl/source/detail/os_util.cpp (hunk 4 above)
Comment (jeffhammond) [reply sycl/source/detail/os_util.cpp]: It is important for the preprocessing in free to match that of malloc.  That way, if someone ever refactors the macros, they will not break the code.

**Pull request merged**